install/installation/installation.go

```diff
@@ -221,6 +223,11 @@ func (k KymaInstaller) PrepareInstallation(artifacts Installation) error {
 		return err
 	}
 
+	err = k.deployInstallerForIstallation(artifacts.InstallerCRYaml)
+	if err != nil {
+		return err
+	}
+
```

Comment:
I do not think it is best to use the same function for applying Installer Yaml as well as Installation CR. Also, Installation CR artifact may not be provided (in case of older versions) so we need to handle both cases.
I would suggest doing something like that:
```go
	if artifacts.InstallerCRYaml != "" {
		err := k.deployInstallationCRYaml(artifacts.InstallerCRYaml)
		if err != nil {
			return err
		}
	}
```
And creating two new functions for deploying installation CR - from Yaml and from Go struct - that could look something like that:
```go
func (k KymaInstaller) deployInstallationCRYaml(yamlFile string) error {

	var installationCR v1alpha1.Installation
	err := yaml.NewDecoder(bytes.NewBuffer([]byte(yamlFile))).Decode(&installationCR)
	if err != nil {
		return fmt.Errorf("error decoding installation CR: %s", err.Error())
	}

	return k.deployInstallationCR(&installationCR)
}

func (k KymaInstaller) deployInstallationCR(installationCR *v1alpha1.Installation) error {

	_, found := installationCR.Labels[installationActionLabel]
	if found {
		delete(installationCR.Labels, installationActionLabel)
	}

	k.infof("Applying Installation CR modifications...")
	k.installationOptions.installationCRModificationFunc(installationCR)
	k.infof("Applying Installation CR...")
	err := k.applyInstallationCR(installationCR)
	if err != nil {
		return fmt.Errorf("failed to apply Installation resources: %w", err)
	}
	k.infof("Installer CR deployed.")
	return nil
}
```

Then we can simplify the  `deployInstaller` function, but still, use the extraction functionality for older versions:
```go
func (k KymaInstaller) deployInstaller(installerYaml string, createResourcesFunc func(resources []k8s.K8sObject) ([]*unstructured.Unstructured, error)) error {
	k.infof("Deploying Installer...")

	k8sInstallerObjects, err := k8s.ParseYamlToK8sObjects(k.decoder, installerYaml)
	if err != nil {
		return fmt.Errorf("failed to parse Installer yaml file to Kubernetes dynamicClientObjects: %w", err)
	}

	// If Installation CR found in installerYaml - apply it otherwise do nothing
	var installationCR *v1alpha1.Installation
	installationCR, k8sInstallerObjects, err = k.extractInstallationCR(k8sInstallerObjects)
	if err != nil {
		if err != errorInstallationCRNotFound {
			return fmt.Errorf("failed to get Installation CR: %w", err)
		}
	}

	_, err = createResourcesFunc(k8sInstallerObjects)
	if err != nil {
		return fmt.Errorf("failed to apply Installer resources: %w", err)
	}

	if installationCR != nil {
		return k.deployInstallationCR(installationCR)
	}

	return nil
}
```